src/sst.rs

```diff
@@ -256,29 +256,62 @@ impl SsTableFormat {
 
     /// Estimate the SsTable size given a memtable, including key, value length
     /// headers, offset ptrs, bloom filter sizes, and block overheads.
+    /// +--------------------------------------------------------------------------------+
+    /// |                                   4K Block                                     |
+    /// +--------------------------------------------------------------------------------+
+    /// |                                                                                |
+    /// | +------------+--------------+----------+---------+----------+------------+     |
+    /// | |   Bloom    |  Number of   |          |   Key   |   Key    |   Value    |     |
+    /// | |   Filter   |  KV Pairs    | Offsets  |  Length |   Data   |   Length   | ... |
+    /// | |            |              |          |         |          |            |     |
+    /// | +------------+--------------+----------+---------+----------+------------+     |
+    /// |                                                                                |
+    /// +--------------------------------------------------------------------------------+
+    /// |                                   4K Block                                     |
+    /// +--------------------------------------------------------------------------------+
+    /// |                                                                                |
+    /// | +------------+--------------+----------+---------+----------+------------+     |
+    /// | |   Bloom    |  Number of   |          |   Key   |   Key    |   Value    |     |
+    /// | |   Filter   |  KV Pairs    | Offsets  |  Length |   Data   |   Length   | ... |
+    /// | |            |              |          |         |          |            |     |
+    /// | +------------+--------------+----------+---------+----------+------------+     |
+    /// |                                                                                |
+    /// +--------------------------------------------------------------------------------+
```

Comment:
❤️

Reply:
Mind including the byte sizes of each in this diagram? And also including a Value Data box?

Reply:
Looking at this diagram more, I think it has a few errors.

1. The bloom filter is stored per-SST not per-block. https://github.com/slatedb/slatedb/blob/04cd8845b870396c6ce785f8157811a760ed275f/src/sst.rs#L445-L454
2. The index should be included, again per-SST. https://github.com/slatedb/slatedb/blob/04cd8845b870396c6ce785f8157811a760ed275f/src/sst.rs#L457-L473
3. An u32 should be included per-SST for checksum. https://github.com/slatedb/slatedb/blob/04cd8845b870396c6ce785f8157811a760ed275f/src/sst.rs#L498
4. Blocks end with a u32 checksum. https://github.com/slatedb/slatedb/blob/04cd8845b870396c6ce785f8157811a760ed275f/src/sst.rs#L436

Also, in the code below, you can use BlockBuilder.estimated_size() to calculate the estimated size for each block.

Reply:
Nice, thanks for the pointers. I had a go at re-organizing the ASCII table and will take a look at the size estimate itself next.